protoutil/blockutils.go

```diff
@@ -218,3 +220,39 @@ func InitBlockMetadata(block *cb.Block) {
 		}
 	}
 }
+
+func VerifyTransactionsAreWellFormed(block *cb.Block) error {
+	if block == nil || block.Data == nil || len(block.Data.Data) == 0 {
+		return fmt.Errorf("empty block")
+	}
+
+	for i, rawTx := range block.Data.Data {
+		env := &cb.Envelope{}
+		if err := proto.Unmarshal(rawTx, env); err != nil {
+			return fmt.Errorf("transaction %d is invalid: %v", i, err)
+		}
+
+		if len(env.Payload) == 0 {
+			return fmt.Errorf("transaction %d has no payload", i)
+		}
+
+		if len(env.Signature) == 0 {
+			return fmt.Errorf("transaction %d has no signature", i)
+		}
```

Comment:
@yacovm I think system channel unit tests are failing in prior release branches because configtxgen doesn't sign the config transaction in the genesis block. Would it be safe to remove this signature check in prior releases?

Reply:
> @yacovm I think system channel unit tests are failing in prior release branches because configtxgen doesn't sign the config transaction in the genesis block. Would it be safe to remove this signature check in prior releases?

For system channel we need a slightly different solution - one that ensures that only a single transaction exists in any block. You can't have 2 transactions in a system channel block, so the problem doesn't manifest vacuously.